File: readability-exercise/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    float Wor = 1;
    float Sen = 0;
    float Let = 0;
    string text = get_string("text: ");
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            Let = Let + 1;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            Sen = Sen + 1;
        }
        else if (isblank(text[i]))
        {
            Wor = Wor + 1;
        }
        // round(math.h)
    }
    // printf("%i, %i, %i", Let, Sen , Wor);
    double L = ((double) Let / Wor) * 100;
    double S = ((double) Sen / Wor) * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    // printf("%i\n", index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n",
               (int) round(index)); // Arredonda o índice para o número inteiro mais próximo
    }
}

File: readability-python/readability.py
from cs50 import get_string

text = get_string("text: ")

Wor = 1
Sen = 0
Let = 0

# checks every caracter in text
for char in text:
    if char.isalnum():
        Let += 1
    elif char.isspace():
        Wor += 1
    elif char in ['.', '?', '!']:
        Sen += 1

L = (Let / Wor) * 100
S = (Sen / Wor) * 100
index = 0.0588 * L - 0.296 * S - 15.8

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade", round(index))
